pkg/libp2p/transport/libp2p.go

```diff
@@ -0,0 +1,203 @@
+package transport
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/bacalhau-project/bacalhau/pkg/compute"
+	pkgconfig "github.com/bacalhau-project/bacalhau/pkg/config"
+	libp2p_host "github.com/bacalhau-project/bacalhau/pkg/libp2p"
+	"github.com/bacalhau-project/bacalhau/pkg/model"
+	"github.com/bacalhau-project/bacalhau/pkg/models"
+	"github.com/bacalhau-project/bacalhau/pkg/pubsub"
+	"github.com/bacalhau-project/bacalhau/pkg/pubsub/libp2p"
+	"github.com/bacalhau-project/bacalhau/pkg/routing"
+	"github.com/bacalhau-project/bacalhau/pkg/system"
+	core_transport "github.com/bacalhau-project/bacalhau/pkg/transport"
+	"github.com/bacalhau-project/bacalhau/pkg/transport/bprotocol"
+	"github.com/hashicorp/go-multierror"
+	libp2p_pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/core/host"
+	basichost "github.com/libp2p/go-libp2p/p2p/host/basic"
+	routedhost "github.com/libp2p/go-libp2p/p2p/host/routed"
+	"github.com/libp2p/go-libp2p/p2p/protocol/identify"
+	"github.com/multiformats/go-multiaddr"
+)
+
+const NodeInfoTopic = "bacalhau-node-info"
+
+type Libp2pTransportConfig struct {
+	Host           host.Host
+	Peers          []string
+	ReconnectDelay time.Duration
+	CleanupManager *system.CleanupManager
+}
+
+type Libp2pTransport struct {
+	Host              host.Host
+	computeProxy      *bprotocol.ComputeProxy
+	callbackProxy     *bprotocol.CallbackProxy
+	nodeInfoPubSub    pubsub.PubSub[models.NodeInfo]
+	nodeInfoDecorator models.NodeInfoDecorator
+}
+
+func NewLibp2pTransport(ctx context.Context,
+	config Libp2pTransportConfig,
+	nodeInfoStore routing.NodeInfoStore) (*Libp2pTransport, error) {
+	// Monkey patch the identify protocol to allow discovering advertised addresses of networks of 3 or more nodes, instead of 5.
+	identify.ActivationThresh = 2
```

Comment:
Comment mentions three or more nodes, but the ActivationThresh is set to 2.  Is this how the ActivationThresh works?

Reply:
yeah this is how it works. Improved the comment in https://github.com/bacalhau-project/bacalhau/pull/3188/commits/425e11d31effd064feb1951d75773e1